src/native/libs/System.Globalization.Native/pal_calendarData.m

```diff
@@ -141,4 +179,123 @@
         return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
     }
 }
+
+/*
+Function:
+GetLatestJapaneseEraNative
+
+Gets the latest era in the Japanese calendar.
+*/
+int32_t GlobalizationNative_GetLatestJapaneseEraNative(void)
+{
+    @autoreleasepool
+    {
+        // Create an NSCalendar with the Japanese calendar identifier
+        NSCalendar *japaneseCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierJapanese];
+        // Get the latest era
+        NSDateComponents *latestEraComponents = [japaneseCalendar components:NSCalendarUnitEra fromDate:[NSDate date]];
+        // Extract the era component
+        NSInteger latestEra = [latestEraComponents era];
+        return (int32_t)latestEra;
+    }
+}
+
+/*
+Function:
+GetJapaneseEraStartDateNative
+
+Gets the starting Gregorian date of the specified Japanese Era.
+*/
+int32_t GlobalizationNative_GetJapaneseEraStartDateNative(int32_t era, int32_t* startYear, int32_t* startMonth, int32_t* startDay)
+{
+    @autoreleasepool
+    {
+        NSCalendar *japaneseCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierJapanese];
+        NSDateComponents *startDateComponents = [[NSDateComponents alloc] init];
+        startDateComponents.era = era;
+        // set the date to Jan 1, 1
+        startDateComponents.month = 1;
+        startDateComponents.day = 1;
+        startDateComponents.year = 1;
+        NSDate *date = [japaneseCalendar dateFromComponents:startDateComponents];
+        NSDate *startDate = date;
+        int32_t currentEra;
+
+        for (int month = 0; month <= 12; month++)
+        {
+            NSDateComponents *eraComponents = [japaneseCalendar components:NSCalendarUnitEra fromDate:date];
+            currentEra = [eraComponents era];
+            if (currentEra == era)
```

Comment:
Is there a chance that `currentEra == era` on the first iteration, but we are more than a month away from the beginning of the era?

The day loop works if we are within a month of entering the desired era (which works if our starting date is before the era since we increase by months to try to get into the right era) with the guarantee that we can step back 31 days to find the start of the era.

Reply:
Month is set to 1 before entering the loop `startDateComponents.month = 1;`, so we always start from January 1.

Reply:
Ah I was missing that we set the calendar to be `year 1` of the desired era, and the start of that era could range from `Jan 1st` to `Dec 31st` on the Gregorian calendar.